File: js/stage1/S1_IF_02.js
// S1_IF_02 — Score to grade
function grade(score) {
  if (score < 0 || score > 100) return null;
  if (score >= 90) return "A";
  if (score >= 80) return "B";
  if (score >= 70) return "C";
  if (score >= 60) return "D";
  return "F";
}
// Tests
[95, 82, 74, 61, 40, -1, 120].forEach((s) => console.log(s, "=>", grade(s)));

File: Python/stage1/S1_IF_02.py
# S1_IF_02 — Score to grade mapping

def grade(score):

    if not isinstance(score, (int, float)) or score < 0 or score > 100:

        return None

    if score >= 90:

        return "A"

    elif score >= 80:

        return "B"

    elif score >= 70:

        return "C"

    elif score >= 60:

        return "D"

    else:

        return "F"


# Boundary tests

tests = [59, 60, 69, 70, 89, 90, 100, 101]

for s in tests:

    print(f"score={s} ->", grade(s))
 